pulsar-common/src/main/java/org/apache/pulsar/common/util/FutureUtil.java

```diff
@@ -69,6 +70,35 @@ public static <T> CompletableFuture<List<T>> waitForAll(Stream<CompletableFuture
                 })));
     }
 
+    /**
+     * Make the dest future complete after another one. {@param dest} is will be completed with the same value as
+     * {@param src}, or be completed with the same error as {@param src}.
+     */
+    public static <T> void completeAfter(final CompletableFuture<T> dest, CompletableFuture<T> src){
+        src.whenComplete((v, ex) -> {
+            if (ex != null){
+                dest.completeExceptionally(ex);
+            } else {
+                dest.complete(v);
+            }
+        });
+    }
+
+    /**
+     * Make the dest future complete after others. {@param dest} is will be completed with the same value as
+     * {@param src} if all the futures of {@param src} is completed, or be completed exceptionally with the same error
+     * as the first one completed exceptionally future of {@param src}.
+     */
+    public static <T> void completeAfterAll(final CompletableFuture<Void> dest, CompletableFuture<T>...src){
+        FutureUtil.waitForAll(Arrays.asList(src)).whenComplete((ignore, ex) -> {
+            if (ex != null){
+                dest.completeExceptionally(ex);
+            } else {
+                dest.complete(null);
+            }
+        });
+    }
```

Comment:
please reformat the code to fix some whitespace formatting issues

Reply:
Formated